src/widgetastic_patternfly/__init__.py

```diff
@@ -1804,3 +1805,118 @@ def items(self):
+    TEXT_BOX = ".//*[@name='{name}' or @id='{id}']"
+    DATE = ".//*[contains(@class, 'datepicker-days')]/table/tbody/tr/*"
+    MONTH = ".//*[contains(@class, 'datepicker-months')]/table/tbody/tr/td/*"
+    YEAR = ".//*[contains(@class, 'datepicker-years')]/table/tbody/tr/td/*"
+    PREV = ".//*[contains(@class, 'prev')]"
+    NEXT = ".//*[contains(@class, 'next')]"
+    DATE_PICKER_SWITH = ".//*[contains(@class, 'datepicker-switch')]"
+
+    def __init__(self, parent, name=None, id=None, logger=None):
+        """Create the widget"""
+        Widget.__init__(self, parent, logger=logger)
+        self.name = name
+        self.id = id
+        self.TEXT_BOX = self.TEXT_BOX.format(name=name, id=id)
+
+    @property
+    def value(self):
+        return self.parent_browser.get_attribute('value', self.TEXT_BOX)
+
+    @property
+    def date_switch_value(self):
+        return self.browser.get_attribute('textContent', self.DATE_PICKER_SWITH)
+
+    def _select_date(self, dd):
+        for el in self.browser.elements(self.DATE):
+            if (not bool({'old', 'new'} & self.browser.classes(el))) and (int(el.text) == int(dd)):
+                el.click()
+                return True
+
+    def _select_month(self, month):
+        for el in self.browser.elements(self.MONTH):
+            if el.text == month:
+                el.click()
+                return True
+
+    def _select_year(self, year):
+        for yr in range(1, 10):
+            start_yr, end_yr = [int(item) for item in self.date_switch_value.split('-')]
+            if start_yr <= int(year) <= end_yr:
+                for el in self.browser.elements(self.YEAR):
+                    if int(el.text) == int(year):
+                        el.click()
+                        return True
+            elif int(year) < start_yr:
+                self.browser.click(self.PREV)
+            elif int(year) > end_yr:
+                self.browser.click(self.NEXT)
+        else:
+            raise ValueError("Year not valid")
+
+    def read(self):
+        return self.value
+
+    def select(self, date):
+        """select date
+
+        Args:
+            date: date in `mm/dd/yyyy` format
+        Returns:
+            :py:class:`bool`
+        """
+        if date == self.value:
+            return False
+
+        self.parent_browser.click(self.TEXT_BOX)
+        date = datetime.strptime(date, "%m/%d/%Y")
+        month = date.strftime("%b")
+        day = date.strftime("%d")
+        year = date.strftime("%Y")
+        sw_value = self.date_switch_value.split(' ')
+
+        if (date.strftime("%B") not in sw_value) or (year not in sw_value):
+            self.browser.click(self.DATE_PICKER_SWITH)
+            if year not in self.date_switch_value:
+                self.browser.click(self.DATE_PICKER_SWITH)
+                self._select_year(year)
+            self._select_month(month)
+        self._select_date(day)
+        return True
+
+    def fill(self, date):
```

Comment:
REQUIRED: please recheck this method. it won't fill-in any values. 
```
<input type="text" name="miq_date_1" id="miq_date_1" value="" readonly="readonly" data-date-orientation="bottom" data-miq_observe_date="{&quot;url&quot;:&quot;/ops/cu_repair_field_changed&quot;}" data-provide="datepicker" data-date-autoclose="true" data-date-format="mm/dd/yyyy" data-date-language="en" data-date-week-start="0">
```
it's read-only

Reply:
@mshriver: I added support for `dict` input with keys `day`, `month` and `year` but I  like to take `str` input as well. So now both will accept. 

while writing test, Most of the time user not going to hardcode any date. In python, he will take help of `datetime` module for date manipulation like In my case, I will try to collect 2-3 days back data so I have to create  `datetime.datetime.now()` object and minus how much days I wish to go back. 
```
from datetime import datetime, timedelta
(datetime.today() - timedelta(days=2)).strftime("%m/%d/%y")
Out[44]: '05/16/18'
```
The point is that if I have `datetime` object no need to send `date`, `month` and `year` in `dirt`.  I can convert into `str` as per `format used by the developer` by checking `widget.date_format` property. So for convenience, I added support for both type input.